python/tank/bootstrap/cached_configuration.py

```diff
@@ -276,6 +291,28 @@ def update_configuration(self):
                     self.path.as_shotgun_dict()
                 )
 
+    def _delete_old_backups(self, backups_folder_path):
+        """
+        Deletes all of the folders in the backups_folder_path
+
+        :param backups_folder_path: File system path to location where core backups
+                                    are temporarily saved
+        """
+    
+        if os.path.exists(backups_folder_path):
+            log.debug("Deleting all core backups in folder: %s" % backups_folder_path)
+            names = os.listdir(backups_folder_path)
+            for name in names:
+                item_path = os.path.join(backups_folder_path, name) 
+                try: 
+                    if os.path.isdir(item_path): 
+                        _make_writeable_recursive(item_path)
+                        shutil.rmtree(item_path)
+                        log.debug("Deleted backups folder: %s" % item_path)
+                except Exception, e: 
+                    log.error("Could not delete %s: %s" % (item_path, e)) 
```

Comment:
if a client for example has a symlink in their config (because they are doing dev, a possible use case!) - this would cause the method to raise, an error is logged and we proceed.

To that note, i suggest we should have a test coverage where we attempt to cleanup, the cleanup fails and we then make sure that the leftovers of this failed cleanup doesn't affect the next attempt when we try to boostrap.